ariadne/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -494,3 +495,46 @@ findFirstUnique lastIdx paths = fromMaybe (error "Can't find a unique path!")
     . head $ dropWhile (`S.member` pathsSet) [lastIdx..]
   where
     pathsSet = V.foldr S.insert S.empty paths
+
+-- ^ This function derives a 3-level address using account index, change index
+--   and address index. The input key is should be the master key (not the key
+--   that was derived from purpose and coin type)
+deriveLvl3KeyPair ::
+       IsBootstrapEraAddr
+    -> PassPhrase
+    -> EncryptedSecretKey
+    -> HdAccountIx
+    -> HdAddressChain
+    -> HdAddressIx
+    -> Maybe (Address, EncryptedSecretKey)
+deriveLvl3KeyPair era pp sk (HdAccountIx accountIdx) change (HdAddressIx addressIdx) =
+    derivePathKeyPair era pp sk
+        [ firstHardened + purpose
+        , firstHardened + coinType
+        , accountIdx
```

Comment:
This should be `firstHardened + accountIdx`.

Reply:
I thought we had discussed this, but perhaps I misunderstood, but okay. I still disagree that this should be the case.